src/uk/co/llobile/kotlin/hours/types/FunctionalOption.kt

```diff
@@ -0,0 +1,66 @@
+package uk.co.llobile.kotlin.hours.types
+
+sealed class FunctionalOption<T>: Functor<T>, Applicative<T>, Monad<T> {
```

Comment:
If an Applicative<T> is also a Functor<T> we don't need to use both of them here.

Reply:
I've made Applicative not extend Functor, as I think that's more correct.